packages/extension/src/newtab/ShortcutLinks/experiments/ShortcutLinksUIV1.tsx

```diff
@@ -0,0 +1,212 @@
+import classNames from 'classnames';
+import { combinedClicks } from '@dailydotdev/shared/src/lib/click';
+import { ActionType } from '@dailydotdev/shared/src/graphql/actions';
+import { cloudinary } from '@dailydotdev/shared/src/lib/image';
+import { useThemedAsset } from '@dailydotdev/shared/src/hooks/utils';
+
+const pixelRatio = globalThis?.window.devicePixelRatio ?? 1;
+const iconSize = Math.round(24 * pixelRatio);
+
+type ShortcutLinksV1Props = {
+  onLinkClick: () => void;
+  onMenuClick: React.MouseEventHandler<Element>;
+  onOptionsOpen: () => void;
+  onV1Hide: () => void;
+  shortcutLinks: string[];
+  shouldUseListFeedLayout: boolean;
+  showTopSites: boolean;
+  toggleShowTopSites: () => void;
+};
+
+const ShortCutV1Placeholder: FC<{
+  initialItem?: boolean;
+  onClick: () => void;
+}> = ({ initialItem = false, onClick }) => {
+  return (
+    <button
+      className="group mr-4 flex flex-col items-center first:mr-2 last-of-type:mr-2 hover:cursor-pointer"
+      onClick={onClick}
+      type="button"
+    >
+      <div className="mb-2 flex size-12 items-center justify-center rounded-full bg-surface-float text-text-secondary group-first:mb-1 group-hover:text-text-primary">
+        <PlusIcon
+          className="inline group-hover:hidden"
+          size={IconSize.Size16}
+        />
+        <PlusIcon
+          className="hidden group-hover:inline"
+          secondary
+          size={IconSize.Size16}
+        />
+      </div>
+      {initialItem ? (
+        <span className="text-text-tertiary typo-caption2">Add shortcut</span>
+      ) : (
+        <span className="h-2 w-10 rounded-10 bg-surface-float" />
+      )}
+    </button>
+  );
+};
+
+const ShortcutV1Item: FC<{
+  url: string;
+  onLinkClick: () => void;
+}> = ({ url, onLinkClick }) => {
+  const cleanUrl = url.replace(/http(s)?(:)?(\/\/)?|(\/\/)?(www\.)?/g, '');
+  return (
+    <a
+      href={url}
+      rel="noopener noreferrer"
+      {...combinedClicks(onLinkClick)}
+      className="group mr-4 flex flex-col items-center"
+    >
+      <div className="mb-2 flex size-12 items-center justify-center rounded-full bg-surface-float text-text-secondary">
+        <img
+          src={`https://api.daily.dev/icon?url=${encodeURIComponent(
+            url,
+          )}&size=${iconSize}`}
+          alt={url}
+          className="size-6"
+        />
+      </div>
+      <span className="max-w-12 truncate text-text-tertiary typo-caption2">
+        {cleanUrl}
+      </span>
+    </a>
+  );
+};
+
+const ShortcutUIItemPlaceholder: FC<PropsWithChildren> = ({ children }) => {
+  return (
+    <div className="group flex flex-col items-center">
+      <div className="mb-2 flex size-12 items-center justify-center rounded-full bg-surface-float text-text-secondary">
+        {children}
+      </div>
+      <span className="h-2 w-10 rounded-10 bg-surface-float" />
+    </div>
+  );
+};
+
+export const ShortcutLinksUIV1: FC<ShortcutLinksV1Props> = (props) => {
```

Comment:
This one is now missing all the tracking events basically.
We'll need to probably find a way to add a hook that has all actions as they are shared between the two.

Reply:
I've passed all the functions from the parent isolating the UI from logic but they should be still running (like in the attached sample). I'm not fully aware on how the events works, do I need to use them in the same component?

<img width="1243" alt="image" src="https://github.com/dailydotdev/apps/assets/4140272/f063f5a8-b536-4c65-9be5-760727c43e33">

Reply:
nope sorry my bad, missed that part and thought you rendered them individual now 😅